ddtrace/internal/writer/writer.py

```diff
@@ -371,7 +371,7 @@ def flush_queue(self, raise_exc: bool = False):
     def _flush_queue_with_client(self, client: WriterClientBase, raise_exc: bool = False) -> None:
         n_traces = len(client.encoder)
```

Comment:
Do we still want this line?

Reply:
No but removing this line causes a test causes mocking in a test to fail. Not sure why. Instead of skipping the test I chose to keep this line as is. I will try to remove it and fix the test in a follow up.

Reply:
I figured out why this test fails: https://github.com/DataDog/dd-trace-py/blob/fc5b2df84c8361eace283bceb53f2be7d7b70ed7/tests/tracer/test_writer.py#L328. 

Here n_traces records the size of the encoder buffer before encoding occurs. If encoding raises an Exception then `client.encoder.encode()` does not return anything. The [encoder.dropped.traces](https://github.com/DataDog/dd-trace-py/blob/fc5b2df84c8361eace283bceb53f2be7d7b70ed7/ddtrace/internal/writer/writer.py#L379) metric is computed using `n_traces` . This metric will be impacted by the same race condition this PR attempts to resolve for the x-datadog-trace-count header but I guess it's not the biggest deal 😅.

Reply:
Understood, could we make a to fix comment above this so it's not lost?